[blockly] contextual info new state of item cannot be interpreted as QuantityType

## The problem

Edit: problem is somewhere else, title adjusted and problem description is https://github.com/openhab/openhab-webui/issues/1942#issuecomment-1611078320.

The current quantity computation block only allows 2 arguments with the same dimension.

## Your suggestion

Ideally, for division and multiplication, it should be possible to have a decimal value as a second argument. It is a common use case to multiply/divide a quantity with a constant or calculated value.

## Additional information

As a bonus (but not critical), it would be nice to align the symbols for multiplication and division with the standard ones in the math calculation block.
The quantity computation block uses * and /. The standard block uses × and ÷.

Here is the blockly code with the problem:

<img width="501" alt="image" src="https://github.com/openhab/openhab-webui/assets/23501634/ea948342-4a1d-4cfe-80d6-a2afd23c56d4">

Generated js:

```
var pveTodayGS2, pveTodayGS1;


console.info(('New PVETodayGS2: ' + String(event.itemState)));
pveTodayGS2 = event.itemState.quantityState;
console.info(('New PVETodayGS2 Qty: ' + String(pveTodayGS2)));
pveTodayGS1 = (pveTodayGS2.quantityState.multiply((cache.shared.get('RATIO')) * 1));
console.info(('Setting PVETodayGS1 to: ' + String(pveTodayGS1)));
```

And extract from the log failure:

```
2023-06-28 11:25:22.079 [INFO ] [nhab.automation.script.ui.778df0dc03] - New PVETodayGS2: 3.459 kWh
2023-06-28 11:25:22.081 [INFO ] [nhab.automation.script.ui.778df0dc03] - New PVETodayGS2 Qty: undefined
2023-06-28 11:25:22.083 [ERROR] [b.automation.script.javascript.stack] - Failed to execute script:
org.graalvm.polyglot.PolyglotException: TypeError: Cannot read property "quantityState" from undefined
	at <js>.:program(<eval>:7) ~[?:?]
	at org.graalvm.polyglot.Context.eval(Context.java:399) ~[?:?]
	at com.oracle.truffle.js.scriptengine.GraalJSScriptEngine.eval(GraalJSScriptEngine.java:458) ~[?:?]
	at com.oracle.truffle.js.scriptengine.GraalJSScriptEngine.eval(GraalJSScriptEngine.java:426) ~[?:?]
	at javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:262) ~[java.scripting:?]
	at org.openhab.automation.jsscripting.internal.scriptengine.DelegatingScriptEngineWithInvocableAndAutocloseable.eval(DelegatingScriptEngineWithInvocableAndAutocloseable.java:53) ~[?:?]
	at org.openhab.automation.jsscripting.internal.scriptengine.InvocationInterceptingScriptEngineWithInvocableAndAutoCloseable.eval(InvocationInterceptingScriptEngineWithInvocableAndAutoCloseable.java:78) ~[?:?]
	at org.openhab.automation.jsscripting.internal.scriptengine.DelegatingScriptEngineWithInvocableAndAutocloseable.eval(DelegatingScriptEngineWithInvocableAndAutocloseable.java:53) ~[?:?]
	at org.openhab.automation.jsscripting.internal.scriptengine.InvocationInterceptingScriptEngineWithInvocableAndAutoCloseable.eval(InvocationInterceptingScriptEngineWithInvocableAndAutoCloseable.java:78) ~[?:?]
	at org.openhab.core.automation.module.script.internal.handler.ScriptActionHandler.lambda$0(ScriptActionHandler.java:71) ~[?:?]
	at java.util.Optional.ifPresent(Optional.java:178) ~[?:?]
	at org.openhab.core.automation.module.script.internal.handler.ScriptActionHandler.execute(ScriptActionHandler.java:68) ~[?:?]
	at org.openhab.core.automation.internal.RuleEngineImpl.executeActions(RuleEngineImpl.java:1175) ~[?:?]
	at org.openhab.core.automation.internal.RuleEngineImpl.runRule(RuleEngineImpl.java:984) ~[?:?]
	at org.openhab.core.automation.internal.TriggerHandlerCallbackImpl$TriggerData.run(TriggerHandlerCallbackImpl.java:87) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
2023-06-28 11:25:22.091 [ERROR] [internal.handler.ScriptActionHandler] - Script execution of rule with UID '778df0dc03' failed: org.graalvm.polyglot.PolyglotException: TypeError: Cannot read property "quantityState" from undefined
```

--- a/bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-scripts.js
+++ b/bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-scripts.js
@@ -173,6 +173,7 @@ export default function defineOHBlocks_Scripts (f7, isGraalJs, scripts) {
           ['triggered event', 'event']
         ]),
         'contextInfo')
+      this.contextInfo = this.getFieldValue('contextInfo')
       this.setInputsInline(true)
       this.setOutput(true, null)
       this.setColour(0)
@@ -192,6 +193,19 @@ export default function defineOHBlocks_Scripts (f7, isGraalJs, scripts) {
         return TIP[contextData]
       })
       this.setHelpUrl('https://www.openhab.org/docs/configuration/blockly/rules-blockly-run-and-process.html#retrieve-rule-context-information')
+    },
+    onchange: function (event) {
+      let contextInfo = this.getFieldValue('contextInfo')
+      if (this.contextInfo !== contextInfo) {
+        this.contextInfo = contextInfo
+        if (contextInfo === 'itemName') {
+          this.setOutput(true, 'oh_item')
+          console.log('type = oh_item')
+        } else {
+          this.setOutput(true, 'String')
+          console.log('type = State String')
+        }
+      }
     }
   }
 

--- a/bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-uom.js
+++ b/bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-uom.js
@@ -195,8 +195,8 @@ function generateQuantityCode (block, inputName) {
       code = `Quantity(${input})`
       break
     case 'oh_itemtype':
-    case '': // vars are expected to contain an item object
-      code = `${input}.quantityState`
+    case '': // vars are expected to contain an item object or the state itself
+      code = `Quantity(${input})`
       break
     case 'oh_item':
       code = `items.getItem(${input}).quantityState`

